types/src/protocol_version.rs

```diff
@@ -143,6 +146,36 @@ impl FromBytes for ProtocolVersion {
     }
 }
 
+impl Serialize for ProtocolVersion {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let str = format!("{}.{}.{}", self.0.major, self.0.minor, self.0.patch);
+        String::serialize(&str, serializer)
```

Comment:
I think we shouldn't force the stringified version for non-human-readable encoders:

```suggestion
        if serializer.is_human_readable() {
            self.0.to_string().serialize(serializer)
        } else {
            self.0.serialize(serializer)
        }
```

Reply:
Done in [`dbe8a60` (#926)](https://github.com/CasperLabs/casper-node/pull/926/commits/dbe8a601912d8ffb4ffb7eec8c9e7d191e92ba6e).